src/components/table/mixins/sort.js

```diff
@@ -0,0 +1,62 @@
+import { EVENT_THEAD_CLICKED } from "../helpers/constants"
+import "./sort.scss"
+
+const SORT_DIRECTION_ASC = "asc"
+const SORT_DIRECTION_DESC = "desc"
+
+export default {
+  data() {
+    return {
+      sortBy: null,
+      sortDir: SORT_DIRECTION_ASC,
+    }
+  },
+  created() {
+    if (this.isSortable) {
+      this.$on(EVENT_THEAD_CLICKED, this.handleSort)
+    }
+  },
+  computed: {
+    sortedData() {
+      const data = [...this.data]
+      return data.sort((a, b) => {
+        const modifier = this.sortDir === SORT_DIRECTION_ASC ? 1 : -1
+        if (a[this.sortBy] < b[this.sortBy]) return -1 * modifier
+        if (a[this.sortBy] > b[this.sortBy]) return 1 * modifier
+        return 0
+      })
+    },
+    isSortable() {
+      return this.fields.some(f => f.sortable)
+    },
+  },
+  methods: {
+    fieldIsSortable(field) {
+      const { sortable } = field
+      return !!sortable
+    },
+    handleSort(field) {
+      if (!this.fieldIsSortable(field)) {
+        return
+      }
+
+      if (field.name === this.sortBy) {
+        this.sortDir =
```

Comment:
Add a return after assigning `this.sortDir`. The `this.sortBy` already is `field.name` anyway, so I consider this a good use case for early returns.